lib/livebook/hubs/personal.ex

```diff
@@ -131,4 +147,54 @@ defimpl Livebook.Hubs.Provider, for: Livebook.Hubs.Personal do
   end
 
   def connection_error(_personal), do: raise("not implemented")
+
+  def notebook_stamp(_hub, _notebook_source, metadata) when metadata == %{} do
```

Comment:
We will only skip if metadata == %{} and the hub is personal. In other words, the criteria for skipping based on metadata should be the hub responsibility.

Reply:
Correct, that's the personal implementation :)